### PR TITLE
Generate the Documentation site using the latest release of SPIRE

### PR DESCRIPTION
**Pull request by mchurichi**
This PR introduces a set of changes to the docs site to keep the content in sync with the latest release of SPIRE.

As a content creator you will still be able to pull markdown file from external repositories, but now if you are pulling from the official SPIRE repo the pull-script will make sure you are always getting it from the Git tag that corresponds to the latest release of SPIRE.
This is done using the GitHub Latest Release API: https://api.github.com/repos/spiffe/spire/releases/latest.

The latest release version is shown at the top left of the screen, just above the ToC.

Additionally, now the pull-script gets the latest release data only once, so it's less likely to hit the quota limit of the GitHub API.

Keep in mind that once a new version of SPIRE is released, the site still need to be rebuild in order to get the info from the new latest release.

Even though this PR is not intended to solve #201, it may be step forward in that direction, and it should help to avoid issues like the one described in #200.

Preview site here: https://deploy-preview-214--spiffe.netlify.app/docs/latest/spiffe-about/overview/

### CODE REVIEW

File: pull_external.py [hunk 1]
```diff
@@ -115,6 +165,7 @@ def _pull_files(yaml_external: dict) -> List[str]:
             )
 
     for target_dir, content in yaml_external.items():
```
Comment (Andres-GC) [inline pull_external.py:167]: Does it makes sense here to extract the two for loops into separate methods?

File: pull_external.py [hunk 2]
```diff
@@ -105,8 +151,12 @@ def _pull_files(yaml_external: dict) -> List[str]:
     # we need to do this as a prep step before processing each file
     # so we can redirect internally
     for target_dir, content in yaml_external.items():
+        source = content.get("source", "").strip()
+        if source == config.get("spireGitHubUrl"):
+            _checkout_switch(content)
         for rel_file in content.get("pullFiles", []):
-            full_url = "{}/blob/master/{}".format(content.get("source"), rel_file)
+            branch = _get_branch_by_repo_url(source)
+            full_url = "{}/blob/{}/{}".format(content.get("source"), branch, rel_file)
```
[the remark this answers is not in the document]
Comment (mchurichi) [reply pull_external.py:159]: it does! good catch

File: layouts/partials/docs/dashboard-panel.html [hunk 3]
```diff
@@ -1,74 +1,24 @@
 {{ $home       := site.BaseURL }}
 {{ $logo       := site.Params.logos.sidebar | relURL }}
 {{ $spiffeLogo := "/img/logos/spiffe/horizontal/color/spiffe-horizontal-color.png" | relURL }}
-{{ $releases   := site.Data.releases }}
-{{/* $latest     := index $releases 1 */}}
-{{ $latest     := "v0.10.0" }}
-{{ $version    := index (split .File.Path "/") 1 }}
-{{ $isLatest   := or (eq $version $latest) (eq $version "latest") }}
+{{ $latest     := getJSON "https://api.github.com/repos/spiffe/spire/releases/latest" }}
```
[the remark this answers is not in the document]
Comment (mchurichi) [reply layouts/partials/docs/dashboard-panel.html:4]: it is not necessary, we can use `site.Data.releases.latest` instead 👍

File: pull_external.py (hunk 1 above)
Comment (mchurichi) [reply pull_external.py:167]: sure, it will improve readability... I'll do it 👌

**Review by Andres-GC — COMMENTED** [review]
LGTM, just a minor comment that I think is should be fixed before the merging.

File: pull_external.py [hunk 4]
```diff
@@ -97,24 +114,59 @@ def _clone_repos(repos: List[str]):
...
 
+    return links
+
+
+def _process_files(yaml_external: Dict) -> List[str]:
+    files = []
     for target_dir, content in yaml_external.items():
+        source = content.get("source", "").strip()
         pull_files: List[str] = content.get("pullFiles", [])
         repo_owner, repo_name = _get_canonical_repo_from_url(content.get("source"))
```
[the remark this answers is not in the document]
Comment (mchurichi) [reply pull_external.py:171]: oops, good catch!

**Review by Andres-GC — APPROVED** [review]
🌭

**Review by sanderson042 — APPROVED** [review]
👍 Approved -- 🙏 @mchurichi for the fix! 

I tested this by looking at the docs/latest/deploying/spire_server/ file referred to in issue #200 to confirm that for the "spire-server healthcheck" section this change pulls in the version with the `-registrationUDSPath` option and not the `-socketPath` option. I also did some spot checks elsewhere in files pulled into spiffe.io from GitHub.

Let's merge this on Monday.

**Pull request merged**